src/Microsoft.HttpRepl/Commands/ListCommand.cs

```diff
@@ -50,7 +50,7 @@ protected override async Task ExecuteAsync(IShellState shellState, HttpState pro
                         SwaggerUri = programState.SwaggerEndpoint,
                         AllowBaseOverrideBySwagger = false
                     };
-                    await apiConnection.SetupHttpState(programState, performAutoDetect: false, cancellationToken).ConfigureAwait(false);
+                    await apiConnection.SetupHttpState(programState, performAutoDetect: false, persistHeaders: true, persistPath: true, cancellationToken).ConfigureAwait(false);
```

Comment:
Not too familiar with the `auto` flag for the list command, but would it ever make sense to include the options to persist headers / path here?

Reply:
The auto flag basically just checks the swagger/openapi description again before executing the command. This is to make sure its working on the latest info. In case, for instance, you're doing this as part of your inner dev loop, you make a change to an API and you want to test it again, you don't have to connect again (to re-read the swagger/open description) - it does that for you. So for that case, I believe it makes sense to always persist the state - you don't want to lose the state during that process.